server/src/core/sms/sms-provider.ts

```diff
@@ -0,0 +1,39 @@
+import createAtClient = require('africastalking');
+import { SmsService as AtSMSService, SendArgs } from 'africastalking-types';
+import { SMSProvider, SendResult } from './types';
+import { AppError, createMessageDeliveryFailedError, createAtApiError } from '../error';
+import { User } from '../user/types';
+
+export interface AtArgs {
+  username: string,
+  apiKey: string
+};
+
+export class AtSMSProvider implements SMSProvider {
+  private atClient: any;
+  private smses: AtSMSService;
+
+  constructor(args: AtArgs) {
+    this.atClient = createAtClient({ username: args.username, apiKey: args.apiKey });
```

Comment:
I think we the `AtSMSProvider` and `AtPaymentsProvider` should use the same AT Client. But not a big deal.